src/features/chat/types.ts

```diff
@@ -0,0 +1,54 @@
+import { NotificationType, ProgressToken, ProtocolRequestType } from 'vscode-languageserver'
+
+import { ChatItemAction, ChatPrompt, ReferenceTrackerInformation, SourceLink, VoteType } from './constants'
+
+export interface ChatParams {
+    tabId: string
+    prompt: ChatPrompt
+    token?: ProgressToken
+}
+export interface ChatResult {
+    body?: string
+    messageId?: string
+    canBeVoted?: boolean // requires messageId to be filled to show vote thumbs
+    relatedContent?: {
+        title?: string
+        content: SourceLink[]
+    }
+    followUp?: {
+        text?: string
+        options?: ChatItemAction[]
+    }
+    codeReference?: ReferenceTrackerInformation[]
+    token?: ProgressToken
+}
+export const chatRequestType = new ProtocolRequestType<ChatParams, ChatResult, ChatResult, void, void>(
+    'aws/sendChatPrompt'
+)
+
+export const chatProgressNotificationType = new NotificationType<ChatResult>('$/progress')
+
+export type EndChatParams = { tabId: string }
+export type EndChatResult = boolean
+export const endChatRequestType = new ProtocolRequestType<EndChatParams, EndChatResult, never, void, void>(
+    'aws/endChat'
+)
+
+export interface QuickActionParams {
+    tabId: string
+    quickAction: string
+    prompt?: string
```

Comment:
nit: would it make sense to reuse `ChatPrompt` type here?

Reply:
I would argue for the opposite, there is no need to have anything more than a string for the quick action for now. ChatPrompt would add extra fields for no purpose